agent-inject/agent/container_init_sidecar.go

```diff
@@ -19,8 +20,10 @@ func (a *Agent) ContainerInitSidecar() (corev1.Container, error) {
 			ReadOnly:  false,
 		},
 		{
-			Name:      a.ServiceAccountName,
-			MountPath: a.ServiceAccountPath,
+			Name: a.ServiceAccountTokenVolume.Name,
+			// Only mount the token
+			MountPath: path.Join(a.ServiceAccountTokenVolume.MountPath, a.ServiceAccountTokenVolume.TokenPath),
+			SubPath:   a.ServiceAccountTokenVolume.TokenPath,
```

Comment:
When I manually tested this for backwards compatibility errors appeared in the injected containers:

```
2020-07-09T17:25:17.904Z [ERROR] auth.handler: error getting path or data from method: error="error reading JWT with Kubernetes Auth: read /var/run/secrets/kubernetes.io/serviceaccount/token: is a directory" backoff=1.178357559
```

Reply:
I guess a lot has changed since I wrote this - will have a look